src/workloads/execution/UnshardedLookup.yml

```diff
@@ -0,0 +1,88 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  Benchmark a $lookup from one unsharded collection to another in a sharded environment.
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &db test
+    Threads: 1
+    DocumentCount: 1000
+    BatchSize: &batchSize 1000
+    CollectionCount: 2
+    Document:
+      a: {^RandomInt: {min: 1, max: 1000}}
+      b: {^RandomInt: {min: 1, max: 100}}
+      c: {^RandomString: {length: 10000}}
+  - &Nop {Nop: true}
+  - *Nop
+
+- Name: Quiesce
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        fsync: 1
+  - *Nop
+
+- Name: Lookup
+  Type: RunCommand
+  Database: *db
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 10
+    Database: *db
+    Operations:
+    - OperationMetricsName: LookupOneToOne
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{
+            $lookup: {
+              from: "Collection1",
+              let: {a0: "$a"},
+              pipeline: [{
+                $match: {
+                  $expr: {$eq: ["$$a0", "$a"]},
+                }
+              }],
+              as: "joined"
+            }
+          }]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: LookupOneToMany
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{
+            $lookup: {
+              from: "Collection1",
+              let: {b0: "$b"},
+              pipeline: [{
+                $match: {
+                  $expr: {$eq: ["$$b0", "$b"]},
+                }
+              }],
+              as: "joined"
+            }
+          }]
+        cursor: {batchSize: *batchSize}
```

Comment:
I did want to raise the question of result set size, which was discussed on my PR [in this thread](https://github.com/mongodb/genny/pull/491#discussion_r660010303). Here are my quick estimations for this workload.

For the second query: each document size is around 10KB and we expect 10 matches per document on the left. This would give us an expected result set size of at least 100MB. 

For the first query: we expect 1 match per document on the left. The amount of data we expect to see in the result set is 1000 documents on the left plus 1000 documents on the right, each at 10KB = 20MB.

What do you think?

Reply:
Thanks for pointing that out Hana! Personally I'd like to avoid adding many new stages to filter the returned results unless it's really necessary, since they could add significant bulk to the latency (especially $count, which desugars to a group) and if anything changes with the implementation or optimization of those stages, these workload results will also reflect those non-lookup related changes.

Instead, what do you think of decreasing the size of the 'c' field for each document, and slightly increasing the chance for more "hits" - so that each payload going over the network has more variation instead of being either one large doc or nothing?

Reply:
Yes thank you @HanaPearlman, this peculiarity invalidated the original benchmark numbers.

@kaywux I see your point. Although I'm not sure the size of the data set is large enough for the extra aggregation phase to have a meaningful impact on the performance differences. The lookup is still the dominate operation in terms of execution time. But I'm happy to remove the aggregation if you think it's minimizing the difference between the two executions.

Reply:
Yep that would be great, thanks in advance for the changes!